### ISSUE TITLE
Missing async support for Reader._createReadStream()

### ISSUE THREAD

**Issue by dvpritzbuer** [overlookmotel/yauzl-promise#17]
Hi,

I use yauzl-promise v4.0.0 and want to use the yauzl.fromReader() method in combination with azure's blob storage client and have created a custom reader class with the following code:

```typescript
import { BlobClient } from "@azure/storage-blob";
import * as yauzl from "yauzl-promise";

export class MyReader extends yauzl.Reader {
  constructor(private readonly blobClient: BlobClient) {
    super();
  }

  async _createReadStream(start: number, length: number) {
    const response = await this.blobClient.download(start, length);
    if (!response.readableStreamBody) throw new Error("no stream");
    
    return response.readableStreamBody;
  }

  async _read(start: number, length: number) {
    const buffer = await this.blobClient.downloadToBuffer(start, length);
    
    return buffer;
  }
``` 

As you can see in this example, I'm not able to provide a _createReadStream implementation without this function to be async and await the blobClient download.

But then I always get the following exception:
![image](https://github.com/overlookmotel/yauzl-promise/assets/71505298/3280a1d0-8213-489b-a72a-00c9533be25d)

Maybe I use the _createReadStream wrong and there is a way for me to provide this method without being async, but if the yauzl.Reader class would await the _createReadStream function, it would solve my problems.

Every part(at least I need) that uses the _createReadStream function is async anyways and I just had to update the following lines to get my custom reader working:

1) entry.js (add await)
![image](https://github.com/overlookmotel/yauzl-promise/assets/71505298/6ac1d9b7-5362-480c-97c7-0e948f9a066f)

2) reader.js (add await)
![image](https://github.com/overlookmotel/yauzl-promise/assets/71505298/dccfc20e-0156-41d9-99c9-1a28806766f5)

3) reader.js (add async)
![image](https://github.com/overlookmotel/yauzl-promise/assets/71505298/58e6cbc6-53a3-492b-8524-b961216ae560)

Thanks for your help!

**Comment by overlookmotel** [comment]
Yes, `Reader.prototype._createReadStream` is not async. It expects the ReadStream to be created synchronously.

I can see that your use case would be easier if `_createReadStream()` was async, and probably that should be supported, though it'd be a breaking change.

I am crazy busy with work for next month, so won't have time to look at this until July (or later). I'd welcome a PR (with tests), and that would likely get it implemented earlier than it would otherwise.

You can also achieve what you want with the current API with something like this:

```js
import {PassThrough, pipeline} from 'node:stream';

export class MyReader extends yauzl.Reader {
  _createReadStream(start, length) {
    const passThroughStream = new PassThrough();

    (async () => {
      try {
        const response = await this.blobClient.download(start, length);
        if (!response.readableStreamBody) throw new Error("no stream");
        // No need for pipeline callback to do anything.
        // Errors will be emitted on passThroughStream anyway.
        pipeline([response.readableStreamBody, passThroughStream], () => {});
      } catch (err) {
        passThroughStream.emit('error', err);
      }
    })();

    return passThroughStream;
  }

  // Other methods omitted in this example
}
```

That example might not cover all bases as far as error handling is concerned, but hopefully you get the idea.

**Comment by dvpritzbuer** [comment]
Thanks for your quick response.

I tried your workaround with the passThrough stream, but then I get following error: "TypeError: The "chunk" argument must be of type string or an instance of Buffer or Uint8Array. Received an instance of PassThrough".

I'm fine with opening a PR and support you as much as I can to get this topic published.
I'm not sure about the tests and documentation updates, but I guess we can clarify these within the PR based on my changes.
